src/jdk.jshell/share/classes/jdk/jshell/execution/impl/ConsoleImpl.java

```diff
@@ -0,0 +1,445 @@
+                                        (remoteOutput.read() <<  0));
+            }
+            return actualLen;
+        }
+
+        private char[] readChars() throws IOException {
+            int actualLen = readInt();
+            char[] result = new char[actualLen];
+            for (int i = 0; i < actualLen; i++) {
+                result[i] = (char) ((remoteOutput.read() <<  8) |
+                                    (remoteOutput.read() <<  0));
+            }
+            return result;
+        }
+
+        private void sendInt(int data) throws IOException {
+            ConsoleImpl.sendInt(remoteInput, data);
+        }
+
+        private int readInt() throws IOException {
+            return (remoteOutput.read() << 24) |
+                   (remoteOutput.read() << 16) |
+                   (remoteOutput.read() <<  8) |
+                   (remoteOutput.read() <<  0);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public synchronized PrintWriter writer() {
+            if (writer == null) {
+                writer = new PrintWriter(new Writer() {
+                    int i;
+                    @Override
+                    public void write(char[] cbuf, int off, int len) throws IOException {
+                        sendAndReceive(() -> {
+                            remoteInput.write(Task.WRITE_CHARS.ordinal());
+                            sendChars(cbuf, off, len);
+                            return null;
+                        });
+                    }
+
+                    @Override
+                    public void flush() throws IOException {
+                        sendAndReceive(() -> {
+                            remoteInput.write(Task.FLUSH_OUTPUT.ordinal());
+                            remoteOutput.read();
+                            return null;
+                        });
+                    }
+
+                    @Override
+                    public void close() throws IOException {
+                    }
+                });
+            }
+            return writer;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public synchronized Reader reader() {
+            if (reader == null) {
+                reader = new Reader() {
+                    @Override
+                    public int read(char[] cbuf, int off, int len) throws IOException {
+                        return sendAndReceive(() -> {
+                            remoteInput.write(Task.READ_CHARS.ordinal());
+                            return readChars(cbuf, off, len);
+                        });
+                    }
+
+                    @Override
+                    public void close() throws IOException {
+                    }
+                };
+            } return reader;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public JdkConsole format(String fmt, Object ...args) {
```

Comment:
```suggestion
        public JdkConsole format(String fmt, Object ... args) {
```

Reply:
There is normally no space before the `...`